.github/workflows/docker.yaml

```diff
@@ -38,27 +38,24 @@ jobs:
             type=sha,prefix=,format=long
             type=semver,pattern=v{{version}}
             type=semver,pattern=v{{major}}.{{minor}}
-      
-      - name: Don't push to registry if this is a PR
-        if: "${{ github.event_name }}" = "pull_request"
-        run: |
-          echo "Not pushing the image to any container registry as this workflow is running on a pull request"
-          exit 0
 
       - name: Log in to DockerHub
+        if: github.event_name != 'pull_request'
         uses: docker/login-action@v2
         with:
           username: ${{ secrets.DOCKER_HUB_USERNAME }}
           password: ${{ secrets.DOCKER_HUB_TOKEN }}
 
       - name: Log in to GHCR
+        if: github.event_name != 'pull_request'
         uses: docker/login-action@v2
         with:
           registry: ghcr.io
           username: ${{ github.repository_owner }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Build and push all platforms
+        if: github.event_name != 'pull_request'
         uses: docker/build-push-action@v4
```

Comment:
Also re-looking at this, doesn't the build happen here. The goal is to run through the build but not push so this seems counter to what we need.